pep-0674.rst

```diff
@@ -288,6 +260,121 @@ the PyDescrObject structure opaque: PyDescr_NAME() and PyDescr_TYPE()"
+
+In May 2020, the ``Py_TYPE()`` and ``Py_SIZE()`` macros have been
+modified to disallow using them as l-values (`Py_TYPE
+<https://github.com/python/cpython/commit/ad3252bad905d41635bcbb4b76db30d570cf0087>`_,
+`Py_SIZE
+<https://github.com/python/cpython/commit/fe2978b3b940fe2478335e3a2ca5ad22338cdf9c>`_).
+
+In November 2020, the change was `reverted
+<https://github.com/python/cpython/commit/0e2ac21dd4960574e89561243763eabba685296a>`__,
+since it broke too many third party projects.
+
+In June 2021, once most third party projects were updated, a `second
+attempt
+<https://github.com/python/cpython/commit/f3fa63ec75fdbb4a08a10957a5c631bf0c4a5970>`_
+was done, but had to be `reverted again
+<https://github.com/python/cpython/commit/6d518bb3a11f9b16098f45b21a13ebe8f537f045>`__
+, since it broke test_exceptions on Windows.
+
+In September 2021, once `test_exceptions has been fixed
+<https://github.com/python/cpython/commit/fb305092a5d7894b41f122c1a1117b3abf4c567e>`_,
+Py_TYPE() and Py_SIZE() were finally `changed
+<https://github.com/python/cpython/commit/f3fa63ec75fdbb4a08a10957a5c631bf0c4a5970>`_.
+
+In November 2021, this backward incompatible change got a
+`Steering Council exception
+<https://github.com/python/steering-council/issues/79#issuecomment-981153173>`_.
+
+
+Backwards Compatibility
+=======================
+
+The proposed C API changes are backward incompatible on purpose.
+
+This change does not follow the :pep:`387` deprecation process. There is
+no known way to emit a deprecation warning only when a macro is used as
+an l-value, but not when it's used differently (ex: as a r-value).
+
+The following 4 macros are left unchanged to reduce the number of
+affected projects: ``PyDescr_NAME()``, ``PyDescr_TYPE()``,
+``PyList_GET_ITEM()`` and ``PyTuple_GET_ITEM()``.
+
+Top 5000 PyPI
+-------------
+
+On January 27, 2022, a code search on the top 5000 PyPI projects found
+that only 7 projects are affected (0.1%):
+
+* datatable (1.0.0):
+  `pending PR <https://github.com/h2oai/datatable/pull/3231>`__
+* guppy3 (3.1.2): `fixed <https://github.com/zhuyifei1999/guppy3/commit/4cb9fcb5d75327544a6875b6caabfdffb70a7e29>`__
+* pickle5 (0.0.12): backport for Python <= 3.7
+* pysha3 (1.0.2): backport for Python <= 3.5
+* python-snappy (0.6.0):
+  `fixed <https://github.com/andrix/python-snappy/commit/1a539d71d5b1ceaf9a2291f21f686cf53a46d707>`__
+* recordclass (0.17.1):
+  `fixed <https://bitbucket.org/intellimath/recordclass/commits/d20d72fa3cdbdcf96c72941560041460adeecff1>`__
+* scipy (1.7.3): fixed (need to update boost)
+
+All of those projects are fixed (but not released), have a pending fix,
+or should not be fixed (backports).
+
+An additional 23 projects only have to regenerate their Cython code to
+use ``Py_SET_TYPE()`` and ``Py_SET_SIZE()``.
+
+Only 2 macros are used as an l-value: ``Py_TYPE()`` and ``Py_SIZE()``.
+
+Projects already fixed
+----------------------
+
+Fixed projects but not released yet (4):
```

Comment:
Only three are listed here. Did you move one to the "released" category (since there's one extra there)?

Also, FWIW, its unclear to me how this category is different from the seven listed above as "affected", several of which also are fixed but not released yet.

Reply:
Fixed in commit cec7b9f5fd77db15a8e81e11314659a396931aa8.

Reply:
I was wrong, M2Crypto and mecab-python3 are no longer affected by the PEP. I clarified the PEP in commit 0a7241c9e1bd05cd552501c5132eb33d9356ebf9.